### ISSUE TITLE
Can't start redis server. Check logs for details.

### ISSUE THREAD

**Issue by pinxiong** [apache/dubbo#7421]
- [x] I have searched the [issues](https://github.com/apache/dubbo/issues) of this repository and believe that this is not a duplicate.
- [x] I have checked the [FAQ](https://github.com/apache/dubbo/blob/master/FAQ.md) of this repository and believe that this is not a duplicate.

### Environment

* Dubbo version: `3.0.0.preview`
* Operating System: `Mac OS X`
* OS version: `10.16`
* Chip: `Apple M1`
* Arch: `x86_64`
* Java version: `11.0.9`

### Steps to reproduce this issue

There are two ways to reproduce this problem: 

+ Run `mvn clean package` after downloading the source codes, or
+ Run the test cases of [RedisMetadataReportTest](https://github.com/apache/dubbo/blob/3.0-preview/dubbo-metadata/dubbo-metadata-report-redis/src/test/java/org/apache/dubbo/metadata/store/redis/RedisMetadataReportTest.java)

### Expected Result

All test cases run successfully.

### Actual Result

All test cases in `dubbo-metadata-report-redis` failed.

```
[INFO] 
[INFO] --- maven-surefire-plugin:2.22.2:test (default-test) @ dubbo-metadata-report-redis ---
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.apache.dubbo.metadata.store.redis.RedisMetadataReportTest
log4j:WARN No appenders could be found for logger (org.apache.dubbo.common.logger.LoggerFactory).
log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
[ERROR] Tests run: 6, Failures: 0, Errors: 6, Skipped: 0, Time elapsed: 2.036 s <<< FAILURE! - in org.apache.dubbo.metadata.store.redis.RedisMetadataReportTest
[ERROR] testSyncStoreProvider  Time elapsed: 0.85 s  <<< ERROR!
java.lang.RuntimeException: Can't start redis server. Check logs for details.
	at org.apache.dubbo.metadata.store.redis.RedisMetadataReportTest.constructor(RedisMetadataReportTest.java:80)

[ERROR] testSyncStoreConsumer  Time elapsed: 0.14 s  <<< ERROR!
java.lang.RuntimeException: Can't start redis server. Check logs for details.
	at org.apache.dubbo.metadata.store.redis.RedisMetadataReportTest.constructor(RedisMetadataReportTest.java:80)

[ERROR] testAuthRedisMetadata  Time elapsed: 0.233 s  <<< ERROR!
java.lang.RuntimeException: Can't start redis server. Check logs for details.
	at org.apache.dubbo.metadata.store.redis.RedisMetadataReportTest.constructor(RedisMetadataReportTest.java:80)

[ERROR] testAsyncStoreProvider  Time elapsed: 0.429 s  <<< ERROR!
java.lang.RuntimeException: Can't start redis server. Check logs for details.
	at org.apache.dubbo.metadata.store.redis.RedisMetadataReportTest.constructor(RedisMetadataReportTest.java:80)

[ERROR] testWrongAuthRedisMetadata  Time elapsed: 0.152 s  <<< ERROR!
java.lang.RuntimeException: Can't start redis server. Check logs for details.
	at org.apache.dubbo.metadata.store.redis.RedisMetadataReportTest.constructor(RedisMetadataReportTest.java:80)

[ERROR] testAsyncStoreConsumer  Time elapsed: 0.171 s  <<< ERROR!
java.lang.RuntimeException: Can't start redis server. Check logs for details.
	at org.apache.dubbo.metadata.store.redis.RedisMetadataReportTest.constructor(RedisMetadataReportTest.java:80)

[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR]   RedisMetadataReportTest.constructor:80 » Runtime Can't start redis server. Che...
[ERROR]   RedisMetadataReportTest.constructor:80 » Runtime Can't start redis server. Che...
[ERROR]   RedisMetadataReportTest.constructor:80 » Runtime Can't start redis server. Che...
[ERROR]   RedisMetadataReportTest.constructor:80 » Runtime Can't start redis server. Che...
[ERROR]   RedisMetadataReportTest.constructor:80 » Runtime Can't start redis server. Che...
[ERROR]   RedisMetadataReportTest.constructor:80 » Runtime Can't start redis server. Che...
[INFO] 
[ERROR] Tests run: 6, Failures: 0, Errors: 6, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] dubbo-build-tools 1.0.0 ............................ SUCCESS [  1.383 s]
[INFO] dubbo-parent 3.0.0.preview ......................... SUCCESS [09:37 min]
[INFO] dubbo-common 3.0.0.preview ......................... SUCCESS [02:16 min]
[INFO] dubbo-container 3.0.0.preview ...................... SUCCESS [  0.171 s]
[INFO] dubbo-container-api 3.0.0.preview .................. SUCCESS [  1.875 s]
[INFO] dubbo-container-spring 3.0.0.preview ............... SUCCESS [  6.278 s]
[INFO] dubbo-serialization 3.0.0.preview .................. SUCCESS [  0.186 s]
[INFO] dubbo-serialization-api 3.0.0.preview .............. SUCCESS [  1.999 s]
[INFO] dubbo-serialization-hessian2 3.0.0.preview ......... SUCCESS [  1.990 s]
[INFO] dubbo-remoting 3.0.0.preview ....................... SUCCESS [  0.166 s]
[INFO] dubbo-remoting-api 3.0.0.preview ................... SUCCESS [ 21.022 s]
[INFO] dubbo-remoting-http 3.0.0.preview .................. SUCCESS [ 10.111 s]
[INFO] dubbo-remoting-netty 3.0.0.preview ................. SUCCESS [ 29.231 s]
[INFO] dubbo-remoting-zookeeper 3.0.0.preview ............. SUCCESS [ 53.559 s]
[INFO] dubbo-remoting-netty4 3.0.0.preview ................ SUCCESS [ 35.556 s]
[INFO] dubbo-rpc 3.0.0.preview ............................ SUCCESS [  0.151 s]
[INFO] dubbo-rpc-api 3.0.0.preview ........................ SUCCESS [ 27.815 s]
[INFO] dubbo-cluster 3.0.0.preview ........................ SUCCESS [01:33 min]
[INFO] dubbo-serialization-jdk 3.0.0.preview .............. SUCCESS [  2.151 s]
[INFO] dubbo-rpc-dubbo 3.0.0.preview ...................... SUCCESS [01:58 min]
[INFO] dubbo-rpc-injvm 3.0.0.preview ...................... SUCCESS [  9.069 s]
[INFO] dubbo-rpc-rest 3.0.0.preview ....................... SUCCESS [ 22.987 s]
[INFO] dubbo-rpc-grpc 3.0.0.preview ....................... SUCCESS [  4.144 s]
[INFO] dubbo-rpc-triple 3.0.0.preview ..................... SUCCESS [ 11.527 s]
[INFO] dubbo-configcenter 3.0.0.preview ................... SUCCESS [  0.103 s]
[INFO] dubbo-configcenter-zookeeper 3.0.0.preview ......... SUCCESS [ 14.692 s]
[INFO] dubbo-metadata 3.0.0.preview ....................... SUCCESS [  0.108 s]
[INFO] dubbo-metadata-api 3.0.0.preview ................... SUCCESS [ 11.347 s]
[INFO] dubbo-registry 3.0.0.preview ....................... SUCCESS [  0.091 s]
[INFO] dubbo-registry-api 3.0.0.preview ................... SUCCESS [ 18.682 s]
[INFO] dubbo-registry-multicast 3.0.0.preview ............. SUCCESS [ 12.997 s]
[INFO] dubbo-registry-zookeeper 3.0.0.preview ............. SUCCESS [ 19.438 s]
[INFO] dubbo-registry-nacos 3.0.0.preview ................. SUCCESS [  3.715 s]
[INFO] dubbo-registry-multiple 3.0.0.preview .............. SUCCESS [ 19.581 s]
[INFO] dubbo-registry-kubernetes 3.0.0.preview ............ SUCCESS [  7.857 s]
[INFO] dubbo-monitor 3.0.0.preview ........................ SUCCESS [  0.168 s]
[INFO] dubbo-monitor-api 3.0.0.preview .................... SUCCESS [ 12.374 s]
[INFO] dubbo-metadata-report-zookeeper 3.0.0.preview ...... SUCCESS [  3.326 s]
[INFO] dubbo-metadata-report-redis 3.0.0.preview .......... FAILURE [  8.329 s]
[INFO] dubbo-configcenter-nacos 3.0.0.preview ............. SKIPPED
[INFO] dubbo-filter 3.0.0.preview ......................... SKIPPED
[INFO] dubbo-filter-cache 3.0.0.preview ................... SKIPPED
[INFO] dubbo-filter-validation 3.0.0.preview .............. SKIPPED
[INFO] dubbo-config 3.0.0.preview ......................... SKIPPED
[INFO] dubbo-config-api 3.0.0.preview ..................... SKIPPED
[INFO] dubbo-registry-dns 3.0.0.preview ................... SKIPPED
[INFO] dubbo-registry-xds 3.0.0.preview ................... SKIPPED
[INFO] dubbo-monitor-default 3.0.0.preview ................ SKIPPED
[INFO] dubbo-config-spring 3.0.0.preview .................. SKIPPED
[INFO] dubbo-demo 3.0.0.preview ........................... SKIPPED
[INFO] dubbo-demo-interface 3.0.0.preview ................. SKIPPED
[INFO] dubbo-plugin 3.0.0.preview ......................... SKIPPED
[INFO] dubbo-qos 3.0.0.preview ............................ SKIPPED
[INFO] dubbo-demo-xml 3.0.0.preview ....................... SKIPPED
[INFO] dubbo-demo-xml-provider 3.0.0.preview .............. SKIPPED
[INFO] dubbo-demo-xml-consumer 3.0.0.preview .............. SKIPPED
[INFO] dubbo-demo-annotation 3.0.0.preview ................ SKIPPED
[INFO] dubbo-demo-annotation-provider 3.0.0.preview ....... SKIPPED
[INFO] dubbo-demo-annotation-consumer 3.0.0.preview ....... SKIPPED
[INFO] dubbo-demo-api 3.0.0.preview ....................... SKIPPED
[INFO] dubbo-demo-api-provider 3.0.0.preview .............. SKIPPED
[INFO] dubbo-demo-api-consumer 3.0.0.preview .............. SKIPPED
[INFO] dubbo-demo-generic-call 3.0.0.preview .............. SKIPPED
[INFO] dubbo-auth 3.0.0.preview ........................... SKIPPED
[INFO] dubbo-compatible 3.0.0.preview ..................... SKIPPED
[INFO] dubbo-dependencies-bom 3.0.0.preview ............... SKIPPED
[INFO] dubbo-configcenter-apollo 3.0.0.preview ............ SKIPPED
[INFO] dubbo 3.0.0.preview ................................ SKIPPED
[INFO] dubbo-bom 3.0.0.preview ............................ SKIPPED
[INFO] dubbo-distribution 3.0.0.preview ................... SKIPPED
[INFO] dubbo-dependencies-zookeeper 3.0.0.preview ......... SKIPPED
[INFO] dubbo-dependencies 3.0.0.preview ................... SKIPPED
[INFO] dubbo-metadata-definition-protobuf 3.0.0.preview ... SKIPPED
[INFO] dubbo-metadata-processor 3.0.0.preview ............. SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  21:42 min
[INFO] Finished at: 2021-03-21T22:45:21+08:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.2:test (default-test) on project dubbo-metadata-report-redis: There are test failures.
```

### Cause

`dubbo` imports `embedded-redis` for redis related testing, however `embedded-redis` cannot support Apple M1 (refer to: [#123](https://github.com/kstyrc/embedded-redis/pull/123), it sounds that nobody to take care of it, also I'm not sure if this PR can fix this issue.)

```
<dependency>
    <groupId>com.github.kstyrc</groupId>
    <artifactId>embedded-redis</artifactId>
    <scope>test</scope>
</dependency>
```

### Suggestion

+ We can upgrade `embedded-redis` by ourselves (the current version is `0.6`), because there are more than one modules importing  `embedded-redis`, such as `dubbo-metadata`, `dubbo-registry-multiple` and `dubbo-dependencies-bom`
+ The root cause of this problem is that redis cannot support Apple M1 better when the version of redis is not `6.0.10` and `6.2 RC2` (refer to: [#8677](https://github.com/redis/redis/issues/8677) and [#8062](https://github.com/redis/redis/issues/8062) ), so we can update the version of redis in  `embedded-redis` from `2.8.19` to `6.0.10` or `6.2 RC2`

### Note

If you search the solution([#7330](https://github.com/apache/dubbo/pull/7330)) to solve this problem, please ignore, because I had a try but it didn't work.

**Comment by kevinw66** [comment]
See this Issue [#7349](https://github.com/apache/dubbo/issues/7349)
And this pr [#7408](https://github.com/apache/dubbo/pull/7408)

**Comment by pinxiong** [comment]
Thans for your help! I really hope #7408 can be merged in `3.0-preview` branch ASAP.
